Source/Core/Core/AchievementManager.cpp

```diff
@@ -471,15 +471,10 @@ const AchievementManager::Badge& AchievementManager::GetAchievementBadge(Achieve
 const AchievementManager::LeaderboardStatus*
 AchievementManager::GetLeaderboardInfo(AchievementManager::AchievementId leaderboard_id)
 {
-  if (const auto leaderboard_iter = m_leaderboard_map.find(leaderboard_id);
-      leaderboard_iter != m_leaderboard_map.end())
-  {
-    if (leaderboard_iter->second.entries.size() == 0)
-      FetchBoardInfo(leaderboard_id);
-    return &leaderboard_iter->second;
-  }
-
-  return nullptr;
+  auto& leaderboard_status = m_leaderboard_map[leaderboard_id];
+  if (leaderboard_status.entries.size() == 0)
+    FetchBoardInfo(leaderboard_id);
+  return &leaderboard_status;
```

Comment:
The code flow feels wrong to me here. Why would the UI (which is the only place that's calling this function) have a better idea about what leaderboards exist than the actual core?

In fact, now that I look at this more, I think I see a race condition. `GetLeaderboardInfo()` can be called from the UI thread and access `m_leaderboard_map`, and whatever thread that calls `LeaderboardEntriesCallback()` can do the same. This is regardless of your changes here but I think your changes here do make it more likely to happen (since you now have both threads potentially writing, rather than just the callback).

Reply:
I was contemplating either this or populating the leaderboard map at creation time and for some reason thought this path was more similar to what I'd been asked to do in other similar instances. No worries, I do see your point.